tests/failpoints/cases/test_gc_worker.rs

```diff
@@ -438,10 +448,35 @@ fn init_compaction_filter(cluster: &Cluster<ServerCluster>, store_id: u64) {
 
     let sim = cluster.sim.rl();
     let gc_worker = sim.get_gc_worker(store_id);
+    gc_worker.set_feature_gate_verion("5.0.0").unwrap();
+    let kv_engine = cluster.get_engine(store_id);
+    // Building a tablet factory
+    let ops = DbOptions::default();
+    let path = Builder::new()
+        .prefix("test_gc_keys_with_region_info_provider")
+        .tempdir()
```

Comment:
All the usage of tempdir is incorrect. Because when the `Tempdir` object goes out of scope, the directory will be automatically deleted, and the engine data will be lost too.

Reply:
What you said is right.  I have corrected other occurrence for path construction except for prepare_gc(). At prepare_gc() invocation time, engine is ready (passed as one parameter), so the temp dir dropping is ok. It's just a skeleton for factory construction, while the factory won't really create any rocksdb later and is only for storing the registry of <<region_id, suffix_id>, rocksdb> in GC_Context.